astropy/stats/funcs.py

```diff
@@ -761,22 +773,51 @@ def median_absolute_deviation(a, axis=None):
     # See https://github.com/numpy/numpy/issues/7330 why using np.ma.median
     # for normal arrays should not be done (summary: np.ma.median always
     # returns an masked array even if the result should be scalar). (#4658)
-    if isinstance(a, np.ma.MaskedArray):
+    if isinstance(data, np.ma.MaskedArray):
         func = np.ma.median
+        if ignore_nan:
+            data = np.ma.masked_where(np.isnan(data), data)
+        elif (NUMPY_LT_1_10 or not NUMPY_LT_1_12) and np.any(np.isnan(data)):
+            # masked arrays containing NaNs should return NaN, but they only
+            # do in 1.11 because of a bug:
+            # https://github.com/astropy/astropy/pull/5232#issuecomment-264751887
+            # TODO for devs: numpy 1.13 may fix this, in which case the
+            # special-casing can be removed
+            return np.nan
+    elif ignore_nan:
+        if NUMPY_LT_1_10 or axis is not None:
+            # older versions don't have nanmedian, but nanmedian
+            # may be faster than np.ma.median *if* axis is None
+            # https://gist.github.com/MSeifert04/1551e80e1aa85b48c9191e727c38f9a9
```

Comment:
oups :( Good that you noticed that. Would be a shame to make decisions based on a bias benchmark.